### ISSUE TITLE
⚠️ Envoy crashing because of DLP 

### ISSUE THREAD

**Issue by asayah** [solo-io/gloo#5491]
**Describe the bug**
Envoy Crashes because of valid configuration, (DLP + Transformation)

**To Reproduce**
apply the config

 
```
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: vs
  namespace: gloo-system
spec:
  virtualHost:
    options:
      stagedTransformations:
        early:
          requestTransforms:
          - requestTransformation:
              transformationTemplate:
                dynamicMetadataValues:
                - key: request_payload
                  value:
                    text: '{{ body }}'    
          responseTransforms:
          - responseTransformation:
              transformationTemplate:
                dynamicMetadataValues:
                - key: response_payload
                  value:
                    text: '{{ body }}'    


      dlp:
        actions:
        - customAction:
            maskChar: X
            name: test
            percent:
              value: 20
            regexActions:
            - regex: (.*msisdn.*:)(.*)
              subgroup: 2
            - regex: (.*espn.*:)(.*)"
              subgroup: 2
        enabledFor: ALL 
    domains:
    - '*'
    routes:
    - routeAction:
        single:
          upstream:
            name: default-httpbin-8000
            namespace: gloo-system
      options:
        autoHostRewrite: true
```

```
apiVersion: gateway.solo.io/v1
kind: Gateway
metadata:
  annotations:
    meta.helm.sh/release-name: glooe
    meta.helm.sh/release-namespace: gloo-system
  name: gateway-proxy
  namespace: gloo-system
spec:
  bindAddress: '::'
  bindPort: 8080
  httpGateway: {}
  options:
    accessLoggingService:
      accessLog:
      - fileSink:
          jsonFormat:
            req_payload: '%DYNAMIC_METADATA(io.solo.transformation:request_payload)%'
            res_payload: '%DYNAMIC_METADATA(io.solo.transformation:response_payload)%'
          path: /dev/stdout
  proxyNames:
  - gateway-proxy
  useProxyProto: false
```

**Expected behavior**
Envoy not to crash because of a valid configuration, (not to crash even if the regex is invalid) 

**Additional context**
Gloo 1.8 and 1.9

**Comment by EItanya** [comment]
I was unable to reproduce locally. Here were my steps:

1. `glooctl install gateway`
2. ```

cat <<EOF | kubectl apply -f -
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: vs
  namespace: gloo-system
spec:
  virtualHost:
    options:
      stagedTransformations:
        early:
          requestTransforms:
          - requestTransformation:
              transformationTemplate:
                dynamicMetadataValues:
                - key: request_payload
                  value:
                    text: '{{ body }}'    
          responseTransforms:
          - responseTransformation:
              transformationTemplate:
                dynamicMetadataValues:
                - key: response_payload
                  value:
                    text: '{{ body }}'    


      dlp:
        actions:
        - customAction:
            maskChar: X
            name: test
            percent:
              value: 20
            regexActions:
            - regex: (.*msisdn.*:)(.*)
              subgroup: 2
            - regex: (.*espn.*:)(.*)"
              subgroup: 2
        enabledFor: ALL 
    domains:
    - '*'
    routes:
    - routeAction:
        single:
          upstream:
            name: default-httpbin-8000
            namespace: gloo-system
      options:
        autoHostRewrite: true
---
apiVersion: gateway.solo.io/v1
kind: Gateway
metadata:
  annotations:
    meta.helm.sh/release-name: glooe
    meta.helm.sh/release-namespace: gloo-system
  name: gateway-proxy
  namespace: gloo-system
spec:
  bindAddress: '::'
  bindPort: 8080
  httpGateway: {}
  options:
    accessLoggingService:
      accessLog:
      - fileSink:
          jsonFormat:
            req_payload: '%DYNAMIC_METADATA(io.solo.transformation:request_payload)%'
            res_payload: '%DYNAMIC_METADATA(io.solo.transformation:response_payload)%'
          path: /dev/stdout
  proxyNames:
  - gateway-proxy
  useProxyProto: false
EOF
```
3. `curl -X POST localhost:8080/anything --data '{"msisdn": "4251235678", "espn": "12345678901234"}'`

**Comment by nfuden** [comment]
Fix is in solo's envoy-ee code and has now been backported

**Comment by nfuden** [comment]
Resolved in gloo-ee 1.8.20 and once released 1.9.3

**Comment by nfuden** [comment]
Point of clarification here as previous comment was  unclear and incorrect.

Our enterprise version of envoy has been updated to fix this and gloo-ee 1.8.19 relies on it in its dependencies

Our enterprise version of envoy has been updated to fix this and the unreleased  gloo-ee 1.9.3 relies on it in its dependencies

No matter gloo-ee you are using your envoy version must be updated in lockstep to correctly pick up this fix.
